services/web/server/src/simcore_service_webserver/projects/projects_db_utils.py

```diff
@@ -308,14 +348,14 @@ async def _get_project(
         project_row = await result.first()
 
         if not project_row:
-            raise ProjectNotFoundError(project_uuid)
-
-        # now carefuly check the access rights
-        if only_published is False:
-            check_project_permissions(
-                project_row, user_id, user_groups, check_permissions
+            raise ProjectNotFoundError(
+                project_uuid=project_uuid,
+                search_context=f"{user_id=}, {only_templates=}, {only_published=}, {check_permissions=}",
```

Comment:
I did not see `search_context` defined on the object. This will get captured in the context. Also in the serialised error this will not be displayed.

But why is this a string and not a dictionary? Since I guess you do not display it?
Maybe check if the use is intended like this.

Reply:
You mean deinfed in ProjectNofFoundError? Yes, it is there but it is stored as context_msg string

Yes, in reality it can be both, i.e. either a dict or a str, as soon as it has a __str__ defined.

Here the idea was to provide some context of "what and how where you looking what you did not find?" :-)